integrations/amazon_bedrock/src/haystack_integrations/components/downloaders/s3/s3_downloader.py

```diff
@@ -0,0 +1,239 @@
+        :param max_cache_size: The maximum number of files to cache.
+        :raises AmazonS3ConfigurationError: If the AWS configuration is not correct.
+        """
+        
+        # Set up download directory
+        file_root_path = file_root_path or os.getenv("FILE_ROOT_PATH")
+        
+        if file_root_path is None:
+            raise ValueError("file_root_path is not set. Please set the file_root_path parameter or the FILE_ROOT_PATH environment variable.")
+        
+        self.file_root_path = Path(file_root_path)
+        
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.aws_region_name = aws_region_name
+        self.aws_session_token = aws_session_token
+        self.aws_profile_name = aws_profile_name
+        self.boto3_config = boto3_config
+        self.file_extensions = [e.lower() for e in file_extensions]
+        self.max_workers = max_workers
+        self.max_cache_size = max_cache_size
+        self.storage : S3Storage = None
+
+        def resolve_secret(secret: Optional[Secret]) -> Optional[str]:
+            return secret.resolve_value() if secret else None
+
+        try:
+            self.session = get_aws_session(
+                aws_access_key_id=resolve_secret(aws_access_key_id),
+                aws_secret_access_key=resolve_secret(aws_secret_access_key),
+                aws_session_token=resolve_secret(aws_session_token),
+                aws_region_name=resolve_secret(aws_region_name),
+                aws_profile_name=resolve_secret(aws_profile_name),
+            )
+            config = Config(
+                user_agent_extra="x-client-framework:haystack", **(self.boto3_config if self.boto3_config else {})
+            )
+
+            self._client = self.session.client("s3", config=config)
+        except Exception as exception:
+            msg = (
+                "Could not connect to Amazon S3. Make sure the AWS environment is configured correctly. "
+                "See https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html#configuration"
+            )
+            raise AmazonS3ConfigurationError(msg) from exception
+
+    def warm_up(self) -> None:
+        """Warm up the component by initializing the settings and storage."""
+        if self.storage is None:
+            settings = S3DownloaderSettings.from_env()
+
+            self.file_root_path.mkdir(parents=True, exist_ok=True)
+
+            self.storage = S3Storage(
+                settings.s3_bucket,
+                settings.s3_folder,
+                endpoint_url=settings.aws_endpoint_url,
+                session=self.session,
+            )
+
+    @component.output_types(documents=List[Document])
+    def run(
+        self,
+        documents: Optional[List[Document]] = None,
+    ) -> Dict[str, List[Document]]:
+        """Download S3-linked files and return enriched `Document`s.
```

Comment:
Could we explain a bit more how the S3 download works? What meta information from the Document is used and how are the returned documents enriched?